src/compiler/checker.ts

```diff
@@ -23653,34 +23654,34 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             return Ternary.True;
         }
 
-        function constructorVisibilitiesAreCompatible(sourceSignature: Signature, targetSignature: Signature, reportErrors: boolean) {
+        function relateConstructorVisibilities(sourceSignature: Signature, targetSignature: Signature, reportErrors: boolean): Ternary | undefined {

```

Comment:
Perhaps usage of `Ternary` is slightly questionable here but it's super convenient. This function now tries to give "definitive answer" to whether those signatures relate to each other or not, it's a short-circuiting of sorts. Usually, it won't be able to give such a definitive answer though and it will just return `undefined` so the caller will proceed with relating the parameters etc